source/agent/addons/internalIO/InternalServerWrapper.cc

```diff
@@ -0,0 +1,157 @@
+// Copyright (C) <2021> Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef BUILDING_NODE_EXTENSION
+#define BUILDING_NODE_EXTENSION
+#endif
+
+#include "InternalServerWrapper.h"
+
+using namespace v8;
+
+DEFINE_LOGGER(InternalServer, "InternalServerWrapper");
+
+Nan::Persistent<Function> InternalServer::constructor;
+
+InternalServer::InternalServer() {
+  async_stats_ = new uv_async_t;
+  stats_callback_ = nullptr;
+  uv_async_init(uv_default_loop(), async_stats_, &InternalServer::statsCallback);
+}
+
+static void destroyAsyncHandle(uv_handle_t *handle) {
+  delete handle;
+}
+
+InternalServer::~InternalServer() {
+  boost::mutex::scoped_lock lock(mutex);
+  if (!uv_is_closing(reinterpret_cast<uv_handle_t*>(async_stats_))) {
+    ELOG_DEBUG("Closing Stats handle");
+    uv_close(reinterpret_cast<uv_handle_t*>(async_stats_), destroyAsyncHandle);
+  }
+  async_stats_ = nullptr;
+}
+
+NAN_MODULE_INIT(InternalServer::Init) {
+  // Constructor template
+  Local<FunctionTemplate> tpl = Nan::New<FunctionTemplate>(New);
+  tpl->SetClassName(Nan::New("InternalServer").ToLocalChecked());
+  tpl->InstanceTemplate()->SetInternalFieldCount(1);
+
+  // Prototype
+  Nan::SetPrototypeMethod(tpl, "close", close);
+  Nan::SetPrototypeMethod(tpl, "getListeningPort", getListeningPort);
+  Nan::SetPrototypeMethod(tpl, "addSource", addSource);
+  Nan::SetPrototypeMethod(tpl, "removeSource", removeSource);
+
+  constructor.Reset(Nan::GetFunction(tpl).ToLocalChecked());
+  Nan::Set(target, Nan::New("InternalServer").ToLocalChecked(),
+           Nan::GetFunction(tpl).ToLocalChecked());
+}
+
+NAN_METHOD(InternalServer::New) {
+  if (info.Length() < 3) {
+    Nan::ThrowError("Wrong number of arguments");
+  }
+
+  if (info.IsConstructCall()) {
+    Nan::Utf8String param0(Nan::To<v8::String>(info[0]).ToLocalChecked());
+    std::string protocol = std::string(*param0);
+
+    unsigned int minPort = Nan::To<unsigned int>(info[1]).FromJust();
+    unsigned int maxPort = Nan::To<unsigned int>(info[2]).FromJust();
+
+    InternalServer* obj = new InternalServer();
+    obj->me = new owt_base::InternalServer(
+        protocol, minPort, maxPort, obj);
+    if (info.Length() > 3) {
+      obj->stats_callback_ = new Nan::Callback(info[3].As<Function>());
+    }
+    obj->Wrap(info.This());
+    info.GetReturnValue().Set(info.This());
+  } else {
+    ELOG_WARN("Not construct call");
+  }
+}
+
+NAN_METHOD(InternalServer::close) {
+  InternalServer* obj = ObjectWrap::Unwrap<InternalServer>(info.Holder());
+  owt_base::InternalServer* me = obj->me;
+  delete me;
+  obj->me = nullptr;
+}
+
+NAN_METHOD(InternalServer::getListeningPort) {
```

Comment:
Maybe a dumb question...  Are you supposing the internal server bind to all interfaces available, and you're not providing config in toml to select interface?